test/src/test/java/hudson/console/HyperlinkNoteTest.java

```diff
@@ -72,7 +78,20 @@ public void textWithNewlinesModelHyperlinkNote() throws Exception {
                 containsString(new ModelHyperlinkNote("", 0).extraAttributes()),
                 containsString(">" + noteTextSanitized + "</a>")));
     }
+    
+    
+    @Test
+    public void textWithSingleQuote() throws Exception {
+        FreeStyleProject upstream = r.createFreeStyleProject("upstream");
+        r.createFreeStyleProject("d0wnstr3'am");
+        upstream.getPublishersList().add(new BuildTrigger("d0wnstr3'am", Result.SUCCESS));
+        FreeStyleBuild b = r.buildAndAssertSuccess(upstream);
+        r.waitUntilNoActivity();
+        HtmlPage rsp = r.createWebClient().goTo(b.getUrl()+"console");
+        assertThat(String.valueOf(rsp.getAnchorByText("d0wnstr3'am").click().getWebResponse().getStatusCode()), containsString("200"));
```

Comment:
Also the test is failing because it's not able to look for the link. Maybe you need to use here the escaped text?

Reply:
> Also the test is failing because it's not able to look for the link. Maybe you need to use here the escaped text?

Even without my fix, the anchor text should return non-null value, its the href atrribute that I am trying to fix. When I tried printing all anchors, there was no downstream project anchor here, but it did exist on my system

Reply:
I'm not sure about why it works on your systme. Are you sure you commited and pushed all your changes?

About the test itself, it's quite confusing to be honest. I believe you should split the sentence into two:
- Check the button exists.
- Click and check the downstream job has been executed.

Reply:
> I'm not sure about why it works on your systme. Are you sure you commited and pushed all your changes?
> 
> About the test itself, it's quite confusing to be honest. I believe you should split the sentence into two:
> 
> * Check the button exists.
> * Click and check the downstream job has been executed.

Sure thing. I will update it with more clarity.

Reply:
@Absh-Day did you have some time to update this?

Reply:
> @Absh-Day did you have some time to update this?

I have modified the test, I am trying finding the console output of the parent build and want to see if it triggered downstream build by checking the existence of text "Triggering a new build of".

Reply:
@varyvol adding r.jenkins.rebuildDependencyGraph(); worked.